Objects

PenguinPaladin opened the pull request.

PenguinPaladin committed:
modified:   src/Main.java
	deleted:    src/homework.java

PenguinPaladin committed:
modified:   src/Main.java

PenguinPaladin committed:
modified:   src/Main.java

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,142 +1,162 @@
 
+    @Override
+    public void checkEngine() {
+        // Велосипеду не требуется проверка двигателя
```
ага, у велосипеда в итоге появился двигатель :D

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,142 +1,162 @@
+// Интерфейс для всех типов транспортных средств
+interface Vehicle {
```
во-первых, все классы/интерфейсы должны быть каждый в отдельном классе. Во-вторых, интерфейс у тебя обязывает все ТС иметь двигатель, что неверно.

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,142 +1,162 @@
+    void updateTyre();
+    void checkEngine();
+}
 
```
класс сервисной станции нужен, зря ты его удалил

dmitry-bizin replied:
![image](https://github.com/CoolNickname68/Education/assets/15345076/729d37ac-50ce-4374-8c34-6c98dff3bd96)

dmitry-bizin replied:
вот схема-подсказка, как нужно реализовать классы, чтобы был полиморфизм

PenguinPaladin committed:
modified:   src/Main.java
	new file:   src/ineterface.java
	new file:   src/vehicles.java

PenguinPaladin committed:
new file:   src/Bicycle.java
	new file:   src/Car.java
	modified:   src/Main.java
	new file:   src/ServiceStation.java
	new file:   src/Truck.java
	renamed:    src/ineterface.java -> src/Vehicle.java
	deleted:    src/vehicles.java

dmitry-bizin commented on src/Vehicle.java:
```diff
@@ -0,0 +1,8 @@
+// Интерфейс для всех типов транспортных средств
+public interface Vehicle {
+    String getModelName();
+    void setModelName(String modelName);
+    int getWheelsCount();
+    void setWheelsCount(int wheelsCount);
+    void updateTyre();
```
здесь должен быть 1 метод (см. схему)

dmitry-bizin commented on src/ServiceStation.java:
```diff
@@ -0,0 +1,17 @@
+// Класс для сервисной станции
+public class ServiceStation {
+    public void check(Vehicle vehicle) {
+        System.out.println("Обслуживаем " + vehicle.getModelName());
+        for (int i = 0; i < vehicle.getWheelsCount(); i++) {
+            vehicle.updateTyre();
+        }
+        if (vehicle instanceof Truck) {
+            ((Truck) vehicle).checkEngine();
+            ((Truck) vehicle).checkTrailer();
+        } else if (vehicle instanceof Car) {
+            ((Car) vehicle).checkEngine();
+        }
+    }
```
в методе будет 1 строчка всего

dmitry-bizin commented on src/Car.java:
```diff
@@ -0,0 +1,34 @@
+// Класс для автомобилей
+public class Car implements Vehicle {
+    private String modelName;
+    private int wheelsCount;
```
поля надо вынести в базовый класс

dmitry-bizin commented on src/Bicycle.java:
```diff
@@ -0,0 +1,30 @@
+// Класс для велосипедов
+public class Bicycle implements Vehicle {
+    private String modelName;
+    private int wheelsCount;
```
поля надо вынести в базовый класс

dmitry-bizin commented on src/Truck.java:
```diff
@@ -0,0 +1,6 @@
+// Класс для грузовиков
+public class Truck extends Car {
```
грузовик НЕ является наследником машины, это неверно. Должен быть отдельный базовый класс Transport с полями, этот класс будут наследовать все 3 вида ТС

PenguinPaladin committed:
modified:   src/Bicycle.java
	modified:   src/Car.java
	modified:   src/Main.java
	modified:   src/ServiceStation.java
	new file:   src/Transport.java
	modified:   src/Truck.java
	modified:   src/Vehicle.java

dmitry-bizin commented on src/Bicycle.java:
```diff
@@ -1,30 +1,14 @@
+        System.out.println("Обслуживаем велосипед " + getModelName());
+        int wheelsCount = getWheelsCount();
+        for (int i = 0; i < wheelsCount; i++) {
+            System.out.println("Меняем покрышку " + (i+1));
```
```suggestion
            System.out.println("Меняем покрышку велосипеда " + (i+1));
```
чтобы подчеркнуть, что это не так, как для машины или грузовика, даже если код один и тот же

dmitry-bizin commented on src/Car.java:
```diff
@@ -1,34 +1,16 @@
+        System.out.println("Обслуживаем автомобиль " + getModelName());
+        int wheelsCount = getWheelsCount();
+        for (int i = 0; i < wheelsCount; i++) {
+            System.out.println("Меняем покрышку " + (i+1));
```
```suggestion
            System.out.println("Меняем покрышку машины " + (i+1));
```
чтобы подчеркнуть, что это не так, как для велосипеда или грузовика, даже если код один и тот же

dmitry-bizin commented on src/Car.java:
```diff
@@ -1,34 +1,16 @@
+        for (int i = 0; i < wheelsCount; i++) {
+            System.out.println("Меняем покрышку " + (i+1));
+        }
+        System.out.println("Проверка двигателя");
```
```suggestion
        System.out.println("Проверка двигателя машины");
```
чтобы подчеркнуть, что это не так, как для грузовика, даже если код один и тот же

dmitry-bizin commented on src/Transport.java:
```diff
@@ -0,0 +1,26 @@
+public class Transport {
```
лучше сделать абстрактным

dmitry-bizin commented on src/Truck.java:
```diff
@@ -1,6 +1,16 @@
+        System.out.println("Обслуживаем грузовик " + getModelName());
+        int wheelsCount = getWheelsCount();
+        for (int i = 0; i < wheelsCount; i++) {
+            System.out.println("Меняем покрышку " + (i+1));
```
```suggestion
            System.out.println("Меняем покрышку грузовика " + (i+1));
```
чтобы подчеркнуть, что это не так, как для машины или велосипеда, даже если код один и тот же

dmitry-bizin commented on src/Truck.java:
```diff
@@ -1,6 +1,16 @@
+        for (int i = 0; i < wheelsCount; i++) {
+            System.out.println("Меняем покрышку " + (i+1));
+        }
+        System.out.println("Проверка двигателя");
```
```suggestion
        System.out.println("Проверка двигателя грузовика");
```
чтобы подчеркнуть, что это не так, как для машины, даже если код один и тот же
